engine/chain/internal/query/query_service.go

```diff
@@ -0,0 +1,106 @@
+package query
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net"
+	"net/http"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/models"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+	"github.com/rs/zerolog"
+)
+
+// Service for querying transactions.
+type Service struct {
+	logger       zerolog.Logger
+	httpClient   *resty.Client
+	url          string
+	token        string
+	deploymentID string
+}
+
+// NewQueryService creates a new query service.
+func NewQueryService(logger zerolog.Logger, httpClient *resty.Client, url, token, deploymentID string) *Service {
+	return &Service{
+		logger:       logger,
+		httpClient:   httpClient,
+		url:          url,
+		token:        token,
+		deploymentID: deploymentID,
+	}
+}
+
+// Query a transaction.
+func (s *Service) Query(ctx context.Context, txID string) (*chainmodels.TXInfo, error) {
```

Comment:
Maybe something like QueryTransaction()?

Reply:
Lets stick to the previous name - I'm gonna rename to `QueryTransaction`